### PR TITLE
[10.0][REF] product_supplierinfo_for_customer: views

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
This PR includes:

-  Refactoring of views, in order to not overwrite views and use inherit and xpath.
-  Only users in group 'product.group_pricelist_item' should be able to view prices in the customer supplierinfo form.

### CODE REVIEW

**Comment by MiquelRForgeFlow** [comment]
@aheficent please review ASAP

File: product_supplierinfo_for_customer/views/product_view.xml [hunk 1]
```diff
@@ -95,18 +69,25 @@
...
+                           context="{'default_search_type':'customer','default_type':'customer','default_product_tmpl_id':id,'form_view_ref':'product_supplierinfo_customer_form_view'}">
+                    </field>
+                </div>
             </div>
         </field>
     </record>
 
+    <record id="product.product_supplierinfo_type_action" model="ir.actions.act_window">
+        <field name="name">Partner Pricelists</field>
+        <field name="res_model">product.supplierinfo</field>
```
Comment (AaronHForgeFlow) [inline product_supplierinfo_for_customer/views/product_view.xml:106]: why is this needed?

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
:-1: purchase pricelist menu requires vendor domain

File: product_supplierinfo_for_customer/views/product_view.xml (hunk 1 above)
Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/views/product_view.xml:106]: to change name of "Vendor Pricelists" to "Partner Pricelists"

Also, I now updated it to pass the supplier domain

Comment (AaronHForgeFlow) [reply product_supplierinfo_for_customer/views/product_view.xml:106]: ok, now it's good

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: thank you for his fix!

**Comment by AaronHForgeFlow** [comment]
ping @jbeficent

**Comment by pedrobaeza** [comment]
In #367 we have also reworked several things for better usability. I'm not sure as I haven't reviewed this if they are in the same wave or not.

**Comment by MiquelRForgeFlow** [comment]
They are in a similar wave, luckily. Even so, if not, I wouldn't accept a "we can't accept this PR because we have already open a new v11 PR with a bit different approach".

**Comment by pedrobaeza** [comment]
Hey, I'm not saying that! I'm saying to exchange ideas for making both the same.

**Comment by MiquelRForgeFlow** [comment]
Ok. In essence, they are the same. They are different only in naming some views. I would prefer merge this PR, and then update the commit history of v11 PR to include this. This way, it would be easier to track differences.

**Comment by pedrobaeza** [comment]
I prefer to keep both as is, as with your option would mean that we have to solve a lot of conflicts. If features of both PRs match, I think it's good to let it as is.

**Comment by MiquelRForgeFlow** [comment]
This PR includes
```python
<xpath expr="//form/group/group[2]" position="attributes">
    <attribute name="groups">product.group_pricelist_item</attribute>
</xpath>
```
An improvement that the v11 PR should include.

**Comment by AaronHForgeFlow** [comment]
:+1: Functional test and working good

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Functional and technical review.
